internal/db/lens.go

```diff
@@ -99,5 +99,82 @@ func (db *DB) setMigration(ctx context.Context, cfg client.LensConfig) (string,
 		return "", err
 	}
 
+	shouldReindex, activeCol, err := db.shouldReindexAfterMigration(ctx, dstCol)
+	if err != nil {
+		return "", err
+	}
+
+	if shouldReindex {
+		err = db.reindexNewActiveVersion(ctx, activeCol)
+		if err != nil {
+			return "", err
+		}
+	}
+
 	return id.String(), nil
 }
+
+// shouldReindexAfterMigration determines if reindexing is needed after adding a migration.
+// Reindexing is needed if:
+// 1. The destination collection is currently active, OR
+// 2. The destination collection is in the history chain of any currently active collection
+// Returns: (shouldReindex bool, activeCollection, error)
+func (db *DB) shouldReindexAfterMigration(
+	ctx context.Context,
+	dstCol client.CollectionVersion,
+) (bool, client.CollectionVersion, error) {
+	if dstCol.IsActive {
+		return true, dstCol, nil
+	}
+
+	activeCol, err := description.GetActiveCollectionByCollectionID(ctx, dstCol.CollectionID)
+	if err != nil {
+		if errors.Is(err, corekv.ErrNotFound) {
+			// No active collection, no reindexing needed
+			return false, client.CollectionVersion{}, nil
+		}
+		return false, client.CollectionVersion{}, err
+	}
+
+	colsWithRoot, err := description.GetCollectionsByCollectionID(ctx, dstCol.CollectionID)
+	if err != nil {
+		return false, client.CollectionVersion{}, err
+	}
+
+	isInChain := isMigrationInActiveChain(dstCol, activeCol, colsWithRoot)
+	return isInChain, activeCol, nil
+}
+
+// isMigrationInActiveChain checks if dstCol (where migration was just added) is in the
+// history chain of the activeCol. This means the migration affects the active version.
+func isMigrationInActiveChain(
```

Comment:
todo: Similar to the other comment, this function looks incorrect and does not take into consideration branching versions.  It also doesn't account for documents that may have been migrated down from higher versions.

If you are not wanting to check for the existence of documents of various versions, I think the only correct way to check for migrations is to get all versions of a collection, and if any of them, regardless of history-location, have a migration, return true.

The same applies to the other comment.